src/Avalonia.X11/XI2Manager.cs

```diff
@@ -256,6 +268,52 @@ private void OnDeviceEvent(IXI2Client client, ParsedDeviceEvent ev)
                     }
                 }
 
+                if(_touchMajorXIValuatorClassInfo is {} touchMajorXIValuatorClassInfo)
+                {
+                    double? touchMajor = null;
+                    double? touchMinor = null;
+                    PixelRect screenBounds = default;
+                    if (ev.Valuators.TryGetValue(touchMajorXIValuatorClassInfo.Number, out var touchMajorValue))
+                    {
+                        Debug.Assert(client is IWindowImpl);
+                        if (client is IWindowImpl windowImpl)
+                        {
+                            var screen = _platform.X11Screens.ScreenFromWindow(windowImpl);
```

Comment:
The window can occupy multiple screens, using ScreenFromWindow for determining the position and dimensions of the touch origin screen isn't sufficient.

Reply:
Thank you @kekekeks . I enlisted the help of my [partner](https://github.com/kkwpsv), borrowed another touchscreen, and assembled a device with two touchscreens. I conducted tests on the Kylin (debian bullseye/sid) system. Through my testing, I found that X11 will automatically scale the input touch TouchMajor across multiple screens based on the screen resolution. Therefore, the Monitor obtained is the Monitor where the touch is located, and the resolution of the Monitor can be used to calculate the value of touch pixels. However, the behavior in X11 is very similar to the situation on Windows without PM DPI awareness enabled. Due to the limitations of X11, my code cannot accurately reflect the physical area on multiple touch screens, but this is a limitation of X11. I suggest waiting until other developers have multi-touch screen requirements in the future before discussing this behavior in detail. At least my current code status can work normally on a single touchscreen.